log_queries.py

```diff
@@ -0,0 +1,78 @@
+import datetime
+
+import pickle
+import os.path
+
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://www.googleapis.com/auth/spreadsheets']  # Allows read and write access
+
+# The ID, range, and auth path for appending to the spreadsheet.
+SPREADSHEET_ID = '15f_zVJxNnUz_9G5ZYPery68rR8PyUnO0yWKSyGQjy4w'
+RANGE_NAME = 'A1'  # Should always place the new query correctly at the bottom of the table
+AUTH_PATH = 'credentials.json'
+
+
+def config_api():
+    """
+    Configures the Google Sheets API service to be used for appending data
+
+    :return: The service as a Resource object, or None if there is no pickle and the authentication JSON can't be found
+    """
+
+    creds = None
+
+    # The file token.pickle stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists('token.pickle'):
+        with open('token.pickle', 'rb') as token:
+            creds = pickle.load(token)
+
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            try:
+                flow = InstalledAppFlow.from_client_secrets_file(AUTH_PATH, SCOPES)
+            except FileNotFoundError as e:
+                print(e)
+                return None
+            creds = flow.run_local_server(port=0)
+
+        # Save the credentials for the next run
+        with open('token.pickle', 'wb') as token:
+            pickle.dump(creds, token)
+
+    return build('sheets', 'v4', credentials=creds)
+
+
+def log_query(service, question: str, answer: str) -> None:
```

Comment:
We also might want to add a positive or negative sentiment association with each question/answer; this isn't implemented in the web chat yet, so it isn't a critical feature (and definitely not required in this PR), but it might be interesting to track which questions get positive sentiment, as well as negative.

Reply:
Ok so are you saying I should add a fourth field for the sentiment, and just automatically fill it with "N/A" or something for now?

Reply:
user review can definitely come in future PR, I agree (good feature to have though).

I do agree w/ Miles that this should return a value / throw an exception

Reply:
@gwholland3, yeah add a fourth field for sentiment and make it N/A for the moment. I agree with what Miles and Jason are saying here.

Reply:
@gwholland3 Can you also add another parameter for the Sheets ID? Leave the constant SPREADSHEET_ID in there and make it an empty string. Then, you could make the parameter default to the constant and @snekiam can pass in the ID as an argument if he decides to go with another method.